pillar/heroku/mitxonline.sls

```diff
@@ -0,0 +1,119 @@
+{% set minion_id = salt.grains.get('id', '') %}
+{% set environment = minion_id.split('-')[-1] %}
+{% set rds_endpoint = salt.boto_rds.get_endpoint('production-apps-rds-postgres-mitxonline') %}
+
+{% set env_dict = {
+    'ci': {
+      'app_name': 'mitxonline-ci',
+      'env_name': 'ci',
+      'GOOGLE_TRACKING_ID': '',
+      'GOOGLE_TAG_MANAGER_ID': '',
+      'release_branch': 'master',
+      'app_log_level': 'INFO',
+      'sentry_log_level': 'ERROR',
+      'OPENEDX_API_BASE_URL': '',
+      'openedx_environment': 'mitxonline-qa',
+      'MAILGUN_FROM_EMAIL': 'MITx Online <no-reply@mitxonline-ci-mail.mitxonline.mit.edu>',
+      'MAILGUN_SENDER_DOMAIN': 'mitxonline-ci-mail.mitxoline.mit.edu',
+      'MITXONLINE_BASE_URL': 'https://mitxonline-ci.mitxonline.mit.edu',
+      'MITXPRO_SECURE_SSL_HOST': 'mitxonline-ci.mitxonline.mit.edu',
+      'vault_env_path': 'rc-apps',
+      },
+    'rc': {
+      'app_name': 'mitxonline-rc',
+      'env_name': 'rc',
+      'GOOGLE_TRACKING_ID': '',
+      'GOOGLE_TAG_MANAGER_ID': '',
+      'release_branch': 'master',
+      'app_log_level': 'INFO',
+      'sentry_log_level': 'ERROR',
+      'OPENEDX_API_BASE_URL': '',
+      'openedx_environment': 'mitxonline-qa',
+      'MAILGUN_FROM_EMAIL': 'MITx Online <no-reply@mitxonline-rc-mail.mitxonline.mit.edu>',
+      'MAILGUN_SENDER_DOMAIN': 'mitxonline-rc-mail.mitxoline.mit.edu',
+      'MITXONLINE_BASE_URL': 'https://mitxonline-rc.mitxonline.mit.edu',
+      'MITXPRO_SECURE_SSL_HOST': 'mitxonline-rc.mitxonline.mit.edu',
+      'vault_env_path': 'rc-apps',
+      },
+    'production': {
+      'app_name': 'mitxonline-production',
+      'env_name': 'ci',
+      'GOOGLE_TRACKING_ID': '',
+      'GOOGLE_TAG_MANAGER_ID': '',
+      'release_branch': 'master',
+      'app_log_level': 'INFO',
+      'sentry_log_level': 'ERROR',
+      'OPENEDX_API_BASE_URL': '',
+      'openedx_environment': 'mitxonline-production',
+      'MAILGUN_FROM_EMAIL': 'MITx Online <no-reply@mitxonline.mit.edu>',
+      'MAILGUN_SENDER_DOMAIN': 'mitxoline.mit.edu',
+      'MITXONLINE_BASE_URL': 'https://mitxonline.mit.edu',
+      'MITXPRO_SECURE_SSL_HOST': 'mitxonline.mit.edu',
+      'vault_env_path': 'production-apps',
+      }
+} %}
+{% set env_data = env_dict[environment] %}
+{% set business_unit = 'mitxonline' %}
+
+proxy:
+  proxytype: heroku
+
+heroku:
+  app_name: {{ env_data.app_name }}
+  api_key: __vault__::secret-{{ business_unit }}/heroku/api_key>data>value
+  config_vars:
+    AWS_ACCESS_KEY_ID:  __vault__:cache:aws-mitx/creds/read-write-delete-xpro-app-{{ env_data.env_name }}>data>access_key
+    AWS_SECRET_ACCESS_KEY: __vault__:cache:aws-mitx/creds/read-write-delete-xpro-app-{{ env_data.env_name }}>data>secret_key
```

Comment:
Copy :spaghetti: 
Should be `mitxonline` rather than `xpro`. Also, we should probably create and manage the buckets and Vault role via Pulumi